src/otx/v2/adapters/torch/mmengine/mmdet/modules/datasets/dataset.py

```diff
@@ -0,0 +1,129 @@
+"""Base MMDataset for Detection Task."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+from mmcv.transforms import Compose
+from mmdet.datasets import BaseDetDataset
+from mmdet.registry import DATASETS
+
+if TYPE_CHECKING:
+    from datumaro.components.dataset import Dataset as DatumDataset
+
+from otx.v2.api.entities.label import LabelEntity
+from otx.v2.api.entities.utils.data_utils import get_old_new_img_indices
+
+
+@DATASETS.register_module()
+class OTXDetDataset(BaseDetDataset):
+    """Wrapper that allows using a OTX dataset to train mmdetection models.
+
+    This wrapper is not based on the filesystem,
+    but instead loads the items here directly from the OTX DatasetEntity object.
+
+    The wrapper overwrites some methods of the BaseDetDataset class: prepare_train_img, prepare_test_img and prepipeline
+    Naming of certain attributes might seem a bit peculiar but this is due to the conventions set in BaseDetDataset. For
+    instance, CustomDatasets expects the dataset items to be stored in the attribute data_infos, which is why it is
+    named like that and not dataset_items.
+
+    """
+
+    class _DataInfoProxy:
+        """This class is intended to be a wrapper to use it in BaseDetDataset-derived class as `self.data_infos`.
+
+        Instead of using list `data_infos` as in BaseDetDataset, our implementation of dataset OTXDataset
+        uses this proxy class with overriden __len__ and __getitem__; this proxy class
+        forwards data access operations to otx_dataset and converts the dataset items to the view
+        convenient for mmdetection.
+
+        Args:
+            otx_dataset (DatasetEntity): DatasetEntity from dataset api
+            labels (List[LabelEntity]): List of LabelEntity
+        """
+
+        def __init__(self, otx_dataset: DatumDataset, labels: list[LabelEntity]) -> None:
+            self.otx_dataset = otx_dataset
+            self.labels = labels
+            self.label_idx = {label.id: i for i, label in enumerate(labels)}
+            self.item_ids = [(item.id, item.subset) for item in self.otx_dataset]
```

Comment:
@wonjuleee I think it'd be better to support indexing in DatumaroDataset. It's a bit confusing that we have to get `self.item_ids`, and call `dataset.get(id=self.item_ids[index][0], subset=self.item_ids[index][1])`